### PR TITLE
Queries

### PR DESCRIPTION
**Pull request by 2004-okari**
### Description

This pull request addresses the following changes and enhancements to the project:

- **Database Schema Modification**: Added a "species" column of type string to the "animals" table in the schema.sql file.

- **Data Insertion**: Added data for new animals in the data.sql file:
  - Charmander, Plantmon, Squirtle, Angemon, Boarmon, Blossom, Ditto.

- **Transactions**: Implemented several transactions in queries.sql, including updates, rollbacks, and commits, to demonstrate various database operations within transactions.

- **Queries**: Added SQL queries to answer questions about the database's contents, such as counting animals, calculating average weights, identifying animals with the most escape attempts, and more.

- **Screenshots**: Included screenshots of the query and transaction results for reference.

### Related Issue

[Link to the related issue or describe the problem this PR solves, if applicable]

### Screenshots

[Attach screenshots to provide visual evidence of the changes made]

### Checklist

- [x] I have tested these changes locally.
- [x] The SQL files have been updated and organized according to the project's conventions.
- [x] The queries and transactions are logically structured and perform the intended tasks.
- [x] The PR has been reviewed and approved by a team member.
- [x] The screenshots accurately reflect the changes made.
- [x] I have updated the README or documentation if necessary.

### Screenshots

Screenshot 2023-09-20 at 12 11 29 PM

Screenshot 2023-09-20 at 11 59 24 AM

Screenshot 2023-09-20 at 12 11 29 PM

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,4 +7,5 @@ CREATE TABLE animals (
     escape_attempts integer,
     neutered boolean,
     weight_kg decimal,
+    species varchar(255)
```
Comment (t-yanick) [inline schema.sql:10]: - [ ] Great job so far. Adding a new column to the table, you are expected to use the SQL command for adding a new column to an existing table. You are expected to use the command below:
`ALTER TABLE animals ADD species varchar(255);`

File: queries.sql [hunk 2]
```diff
@@ -31,3 +31,26 @@ WHERE NOT name = 'Gabumon';
 SELECT * 
 from animals 
 WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
```
Comment (t-yanick) [inline queries.sql:34]: - [ ] As per project requirements, you have missed part of the requirements. I am pasting all what you are missing below. Kindly add this part.
- I see some of the transactions in your screenshots, but I cannot find the commands in the queries.sql file
```
**Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.
Take a screenshot of the results of your actions.
Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction
Take a screenshot of the results of your actions.
Remember to add all these transactions to your queries.sql file.**
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @2004-okari , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All SQL commands put in the correct files
:heavy_check_mark: Screenshots added to PR description
:heavy_check_mark: Use of correct Gitflow

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (2004-okari) [reply schema.sql:10]: I used it, I happen to not have included it in the screenshot

Comment (2004-okari) [reply schema.sql:10]: the screenshots indicate there is a species column.

File: queries.sql [hunk 3]
```diff
@@ -53,4 +53,29 @@ GROUP BY species;
...
+ALTER TABLE animals
+ADD COLUMN species varchar(255);
+
+-- transaction 1
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
+COMIT;
+
```
Comment (mikethreels) [inline queries.sql]: - [x] these are supposed to be 2 separate transactions.
- after setting the species column to unspecified you're supposed to verify if the changes are made
- then you should rollback the changes
- and verify if the changes were undone.

- [x] line 65 and 66 should be in a separate transaction
- after updating the columns you're supposed to verify if the changes are made
- after that commit the changes 
- and after commit check if the changes persist.

File: queries.sql [hunk 4]
```diff
@@ -53,4 +53,29 @@ GROUP BY species;
...
+-- transaction 2
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (mikethreels) [inline queries.sql]: - [x] in this transaction after deleting the table you're supposed to verify if the table got deleted.
- [x] then after rollback verify if it was restored.

File: queries.sql [hunk 5]
```diff
@@ -53,4 +53,29 @@ GROUP BY species;
...
 FROM animals
 WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
-GROUP BY species;
\ No newline at end of file
+GROUP BY species;
+
+-- Alter to add new column
+ALTER TABLE animals
+ADD COLUMN species varchar(255);
+
```
Comment (mikethreels) [inline queries.sql]: - [x] according to the requirements Everything that creates or change the tables structure should be saved in the schema.sql file.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Nyandika, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- good job writing the queries

 ## Required Changes ♻️
### after making the requested changes please make sure to include all the updated screenshots.
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
## Hi @2004-okari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Nice job addressing the comments from the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**